packages/scene-composer/src/augmentations/components/three-fiber/anchor/AnchorWidget.tsx

```diff
@@ -124,11 +128,48 @@ export function AsyncLoadedAnchorWidget({
       WarningIconSvgString,
       ErrorIconSvgString,
       VideoIconSvgString,
+      CustomIconSvgString,
     ];
     return iconStrings.map((iconString, index) => {
+      if (keys[index] === 'Custom') {
```

Comment:
can we move those code into the file for `svgIconToWidgetSprite` and let `svgIconToWidgetSprite` handle it? so here we still only need to do 
```
return svgIconToWidgetSprite(iconString, keys[index], isAlwaysVisible, !autoRescale);
```

Reply:
All I am doing here is to fill the color for selected svg and then sending it to svgIconToWidgetSprite. What's the advantage of svgIconToWidgetSprite handling this logic?

Reply:
I mainly mean those code should be in some utils file not in this component. either AnchorWidget or svgIconToWidgetSprite calls that util to generate the svg are both fine